python3Packages.streamingjson: init at 0.0.5

ethanthoma opened the pull request:
streamingjson is a JSON streaming preprocessor library

## Things done:
- Built on platform:
    - [x] x86_64-linux
    - [ ] aarch64-linux
    - [ ] x86_64-darwin
    - [ ] aarch64-darwin
- Tested, as applicable:
    - [ ] [NixOS tests](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) in [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests).
    - [ ] [Package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests) at passthru.tests.
    - [ ] Tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test) for functions and "core" functionality.
- [ ] Ran nixpkgs-review on this PR. See [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage).
- [ ] Tested basic functionality of all binary files, usually in ./result/bin/.
- Nixpkgs Release Notes
    - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
    - [ ] Module addition: when adding a new NixOS module.
    - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md), [pkgs/README.md](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md), [maintainers/README.md](https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md) and other READMEs.
---
Add a 👍 [reaction](https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to [pull requests you find important](https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc).

acid-bong commented on pkgs/development/python-modules/streamingjson/default.nix:
```diff
@@ -0,0 +1,35 @@
+
+  dependencies = [ ];
+
+  doCheck = false;
```
must specify why:

> If checks are partially or fully disabled, document why.

https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#new-packages (sadly not in a very convenient place of the docs)

acid-bong commented on pkgs/development/python-modules/streamingjson/default.nix:
```diff
@@ -0,0 +1,35 @@
+    homepage = "https://pypi.org/project/streamingjson/";
+    license = licenses.mit;
+    maintainers = with maintainers; [ ethanthoma ];
+  };
```
`with` before attrsets (especially `with lib;`, _especially_ file-wide ones, _**especially**_ multiple ones) has been abused to death in the past and is now frowned upon[^1]. Just prepend `lib.` where necessary

[^1]: `with` before small scopes, like lists, is fine

```suggestion
  meta = {
    description = "Streamlined, user-friendly JSON streaming preprocessor";
    homepage = "https://pypi.org/project/streamingjson/";
    license = lib.licenses.mit;
    maintainers = with lib.maintainers; [ ethanthoma ];
  };
```

acid-bong replied:
no need to reply in each convo, just mark as resolved when it's resolved

ethanthoma committed:
maintainers: add ethanthoma

ethanthoma commented on pkgs/development/python-modules/streamingjson/default.nix:
```diff
@@ -0,0 +1,35 @@
+  version = "0.0.5";
+  pyproject = true;
+
+  disabled = pythonOlder "3.7";
```
ok done!

acid-bong commented on pkgs/development/python-modules/streamingjson/default.nix:
```diff
@@ -0,0 +1,33 @@
+  src = fetchFromGitHub {
+    owner = "karminski";
+    repo = "streaming-json-py";
+    rev = version;
```
```suggestion
    tag = version;
```

acid-bong reviewed (approved):
otherwise looks great, i'll approve already

ethanthoma committed:
python3Packages.streamingjson: init at 0.0.5

ethanthoma commented:
@acid-bong apologies for the ping but do I wait for natsukium to approve? I saw in other python3Packages PRs that isn't typically the case. Thanks!

FliegendeWurst commented on pkgs/development/python-modules/streamingjson/default.nix:
```diff
@@ -0,0 +1,33 @@
+  pname = "streamingjson";
+  version = "0.0.5";
+  pyproject = true;
+
```
```suggestion

  __structuredAttrs = true;

```
This is required for new packages

ethanthoma replied:
done

FliegendeWurst commented on maintainers/maintainer-list.nix:
```diff
@@ -7945,6 +7945,11 @@
       { fingerprint = "2E51 F618 39D1 FA94 7A73  00C2 34C0 4305 D581 DBFE"; }
     ];
   };
+  ethanthoma = {
```
There is a merge conflict here

ethanthoma commented:
rebased on master (fixes the maintainer-list conflict) and added `__structuredAttrs = true`. 0.0.5 is still the latest upstream release. builds locally on x86_64-linux, 7 tests pass.
